examples/trusted_database/module_0/rust/src/lib.rs

```diff
@@ -0,0 +1,102 @@
+//! clients.
+//!
+//! Current example implementation uses a list of Santander Cycles in London as test database:
+//! https://tfl.gov.uk/tfl/syndication/feeds/cycle-hire/livecyclehireupdates.xml
+//!
+//! Trusted Database example consists of two Oak Nodes: Main Node and Handler Node.
+//!
+//! Upon receiving a client request, Main Node creates a new dedicated Handler Node and proxies
+//! the request to it. Main Node also contains the database of Points Of Interest and shares it with
+//! all newly created Handler Nodes.
+//!
+//! The key idea is that clients don't need to trust any particular Wasm module and only need to
+//! trust in the Oak implementation. Thus clients don't need to use module hash or module signature
+//! labels and can only assign their token labels to the data.
+
+mod database;
+#[cfg(test)]
+mod tests;
+
+use database::load_database;
+use log::debug;
+use oak::{grpc, io::SenderExt, proto::oak::invocation::GrpcInvocationReceiver, Node};
+use trusted_database_handler::proto::oak::examples::trusted_database::{
+    PointOfInterest, TrustedDatabaseCommand,
+};
+
+/// Oak Node that contains an in-memory database.
+pub struct TrustedDatabaseNode {
+    points_of_interest: Vec<PointOfInterest>,
+}
+
+impl Node<grpc::Invocation> for TrustedDatabaseNode {
+    fn handle_command(&mut self, invocation: grpc::Invocation) -> Result<(), oak::OakError> {
+        // Create a client request handler Node.
+        debug!("Creating handler Node");
+        let (sender, receiver) = oak::io::channel_create::<TrustedDatabaseCommand>()
+            .expect("Couldn't create command channel");
+        oak::node_create(
+            &oak::node_config::wasm("trusted_database_handler", "handler_oak_main"),
+            receiver.handle,
+        )
+        .expect("Couldn't create handler Node");
+        oak::channel_close(receiver.handle.handle).expect("Couldn't close receiver channel");
+
+        // Create a gRPC invocation channel.
+        let (invocation_sender, invocation_receiver) =
+            oak::io::channel_create::<grpc::Invocation>()
+                .expect("Couldn't create gRPC invocation channel");
+
+        // Create a command message that contains a copy of the database.
+        let command = TrustedDatabaseCommand {
+            invocation_receiver: Some(GrpcInvocationReceiver {
+                receiver: Some(invocation_receiver),
+            }),
+            points_of_interest: self.points_of_interest.to_vec(),
+        };
+
+        // Send a command massage to the handler Node.
+        debug!("Sending command message to handler Node");
+        sender
+            .send(&command)
+            .expect("Couldn't send command to handler Node");
+        oak::channel_close(sender.handle.handle).expect("Couldn't close sender channel");
+
+        // Send a gRPC invocation.
+        debug!("Sending gRPC invocation to handler Node");
```

Comment:
IIUC the "handler node" in this log line is different from the "handler node" in the previous debug line. If so, it is better to use different names to be clear.